core/src/classes/plugin.ts

```diff
@@ -380,9 +382,12 @@ export interface SourceFilterMethod {
     sourceId: string;
     sourceOptions: SimpleSourceOptions;
     sourceMapping: SourceMapping;
-    property: Property;
-    propertyId: string;
-    propertyOptions: SimplePropertyOptions;
+    property?: Property;
+    propertyId?: string;
+    propertyOptions?: SimplePropertyOptions;
+    schedule?: Schedule;
+    scheduleId?: string;
+    scheduleOptions?: SimpleScheduleOptions;
```

Comment:
`SourceFilterMethod` now is generic and can work for both Properties or Schedules.  To date, this method is static, and basically just returns the list of columns in the table